src/indentcodeblockcommand.js

```diff
@@ -0,0 +1,315 @@
+/**
+ * @license Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module code-block/indentcodeblockcommand
+ */
+
+import Command from '@ckeditor/ckeditor5-core/src/command';
+import first from '@ckeditor/ckeditor5-utils/src/first';
+import { getLeadingWhiteSpaces } from './utils';
+
+/**
+ * The code block indentation command plugin.
+ *
+ * @extends module:core/command~Command
+ */
+export default class IndentCodeBlockCommand extends Command {
+	constructor( editor, indentDirection ) {
+		super( editor );
+
+		/**
+		 * Determines whether this command indents (`'forward'`) or outdents (`'backward'`).
+		 *
+		 * @readonly
+		 * @private
+		 * @member {String}
+		 */
+		this._indentDirection = indentDirection;
```

Comment:
WDYT about spiting this to two commands (with some common utils)?

I see that we did such commands as one in the past but I get the feeling more and more that those commands should be separated.

I see from the code that there are cases when `if()` (or `else`) is relatively simple and the other case is trickier and in general the code is harder to read as you have to branch each method in brain.

Also those will always be separate commands in the editor so it does make sense to split them as `IndentCodeBlockCommand` and `OutdentCodeBlockCommand` (love the long names BTW ;) ).

Reply:
That's a good idea. Consider it done.